From 3fbc353d9018735140bf24e93ae35cedcea7eb62 Mon Sep 17 00:00:00 2001
From: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
Date: Wed, 18 Mar 2026 11:20:04 +0200
Subject: [PATCH] doc: workaround removed docutils.error_reporting

This was removed by the kernel doc module still depends on it.
Workaround by providing a shim.

Signed-off-by: Pantelis Antoniou <pantelis.antoniou@konsulko.com>
---
 doc/conf.py | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/doc/conf.py b/doc/conf.py
index 8c3dcd280..c3e9c38dd 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -14,8 +14,27 @@
 import pathlib
 import subprocess
 import sys
+import types
 sys.path.insert(0, os.path.abspath('.'))
 
+# linuxdoc still imports docutils.utils.error_reporting, which was removed in
+# newer docutils releases. Provide the small API surface linuxdoc uses so Sphinx
+# can keep loading with either docutils version.
+try:
+    import docutils.utils.error_reporting  # type: ignore  # noqa: F401
+except ModuleNotFoundError:
+    error_reporting = types.ModuleType('docutils.utils.error_reporting')
+
+    def ErrorString(error):
+        return str(error)
+
+    def SafeString(value):
+        return str(value)
+
+    error_reporting.ErrorString = ErrorString
+    error_reporting.SafeString = SafeString
+    sys.modules['docutils.utils.error_reporting'] = error_reporting
+
 
 # -- Project information -----------------------------------------------------
 
